esp/esp/program/modules/handlers/teachersurveymodule.py

```diff
@@ -78,17 +61,16 @@ def teachers(self, QObject = False):
             return {'teacher_survey': Q(record__program=program) & Q(record__event=event)}
         return {'teacher_survey': ESPUser.objects.filter(record__program=program, record__event=event).distinct()}
 
-    def studentDesc(self):
-        return {'student_survey': """Students who filled out the survey"""}
-
     def teacherDesc(self):
         return {'teacher_survey': """Teachers who filled out the survey"""}
 
     def isStep(self):
-        return False
+        return (Tag.getBooleanTag('teacher_survey_isstep', program=self.program, default=False) and
+                self.program.getTimeSlots()[0].start < datetime.datetime.now() and
+                self.program.getSurveys().filter(category__in = ["learn","teach"]).exists())
```

Comment:
Just a reminder that if you implement my comment about moving the survey results link, then this should just check for a teacher survey